R/mod_export_count_table.R

```diff
@@ -0,0 +1,208 @@
+              )),
+              easyClose = TRUE
+            )
+          )
+        }
+      })
+
+
+      # Add/remove checkbox warning
+      checkbox_label <- shiny::eventReactive(input[[EXP$ID$DATAPROTECT_BOX]], {
+        if (input[[EXP$ID$DATAPROTECT_BOX]]) {
+          return(paste(EXP$MSG$DATAPROTECT_LABEL, intended_use_label))
+        } else {
+          return(shiny::tags$embed(
+            paste(EXP$MSG$DATAPROTECT_LABEL, intended_use_label),
+            shiny::br(shiny::tags$span(
+              style = paste("color:", "#CC3333"),
+              EXP$MSG$DATAPROTECT_WARN
+            ))
+          ))
+        }
+      })
+
+      # Update checkbox label separately (to allow testing of logic)
+      output$label_id <- shiny::renderUI(checkbox_label())
+
+
+      # Check if we should enable/disable download button
+      download_enable <- shiny::eventReactive(
+        c(input[[EXP$ID$FILENAME_BOX]], input[[EXP$ID$DATAPROTECT_BOX]]),
+        {
+          if (
+            (input[[EXP$ID$FILENAME_BOX]] != "") &
+              input[[EXP$ID$DATAPROTECT_BOX]]
+          ) { # nolint end
+            return(TRUE)
+          } else {
+            return(FALSE)
+          }
+        }
+      )
+
+      # Enable/disable download button separately (to allow testing of logic)
+      shiny::observeEvent(download_enable(), {
+        if (download_enable()) {
+          shinyjs::enable(EXP$ID$DOWNLOAD_BUTTON)
+        } else {
+          shinyjs::disable(EXP$ID$DOWNLOAD_BUTTON)
+        }
+      })
+
+      # Download
+      output[[EXP$ID$DOWNLOAD_BUTTON]] <- shiny::downloadHandler(
+        filename = function() {
+          paste0(input[[EXP$ID$FILENAME_BOX]], ".xlsx")
+        },
+        content = function(file) {
+          shiny::removeModal() # close pop up
+
+          # Dataframe contains summary and all subject info. as list of lists
+          # Extract the required named list
+          # Add overall patients from column header
+          # Separate the columns into count and percentage
+
+
+          # Get col names and total patients
+          total_colname <- dataset[["meta"]]$n_denominator
+
+          #Get event variables
+          event_vars <- dataset[["meta"]]$hierarchy
+
+          excelfile <- dataset[["df"]] |>
+            dplyr::select(
+              event_vars,
+              names(total_colname)
+            ) |>
+            dplyr::mutate(dplyr::across(
+              dplyr::where(is.list),
+              ~ purrr::map_chr(., "count")
+            )) |>
+            dplyr::mutate(dplyr::across(
+              dplyr::all_of(names(total_colname)),
+              ~ gsub("\u2014", NA, .)
+            )) |>
+            dplyr::mutate(dplyr::across(
+              dplyr::all_of(event_vars),
+              ~ gsub("\035", "Total", .)
+            ))
+
+          excelfile <- add_total_patient(
+            excelfile,
+            total_colname
+          )
```

Comment:
I would extract this in a separate function, and test is in a non reactive environment. 

Maybe with the following structure

``` r
count_table <- call_table_computing_function(xxx)
call_data_extraction(count_table)
```

This way if we change the way we count in the table we break the test and remember the coupling between both of them